monitoring/logicmonitor.py

```diff
@@ -0,0 +1,2170 @@
+   Copyright (C) 2015  LogicMonitor
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program; if not, write to the Free Software Foundation,
+   Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301  USA"""
+
+import datetime
+import os
+import platform
+import socket
+import sys
+import types
+import urllib
+
+HAS_LIB_JSON = True
+try:
+    import json
+    # Detect the python-json library which is incompatible
+    # Look for simplejson if that's the case
+    try:
+        if (
+         not isinstance(json.loads, types.FunctionType) or
+         not isinstance(json.dumps, types.FunctionType)
+        ):
+            raise ImportError
+    except AttributeError:
+        raise ImportError
+except ImportError:
+    try:
+        import simplejson as json
+    except ImportError:
+        print(
+            '\n{"msg": "Error: ansible requires the stdlib json or ' +
+            'simplejson module, neither was found!", "failed": true}'
+        )
+        HAS_LIB_JSON = False
+    except SyntaxError:
+        print(
+            '\n{"msg": "SyntaxError: probably due to installed simplejson ' +
+            'being for a different python version", "failed": true}'
+        )
+        HAS_LIB_JSON = False
+
+RETURN = '''
+---
+success:
+    description: flag indicating that execution was successful
+    returned: success
+    type: boolean
+    sample: True
+...
+'''
+
+
+DOCUMENTATION = '''
+---
+module: logicmonitor
+short_description: Manage your LogicMonitor account through Ansible Playbooks
+description:
+  - LogicMonitor is a hosted, full-stack, infrastructure monitoring platform.
+  - This module manages hosts, host groups, and collectors within your LogicMonitor account.
+version_added: "2.2"
+author: Ethan Culler-Mayeno, Jeff Wozniak
+notes:
+  - You must have an existing LogicMonitor account for this module to function.
+requirements: ["An existing LogicMonitor account", "Linux"]
+options:
+  target:
+    description:
+      - The type of LogicMonitor object you wish to manage.
+      - "Collector: Perform actions on a LogicMonitor collector"
+      - NOTE You should use Ansible service modules such as 'service' or 'supervisorctl' for managing the Collector 'logicmonitor-agent' and 'logicmonitor-watchdog' services. Specifically, you'll probably want to start these services after a Collector add and stop these services before a Collector remove.
```

Comment:
If you use `M(...)` the documentation generator will put in links to those modules

e.g.
`M(service) or M(supervisorctl)`